tests/test/kube.rs

```diff
@@ -0,0 +1,349 @@
+        }
+    }
+
+    /// Gets a list of resources restricted by the label selector.
+    ///
+    /// The label selector supports `=`, `==`, `!=`, and can be comma
+    /// separated: `key1=value1,key2=value2`.
+    pub fn list_labeled<K>(&self, label_selector: &str) -> ObjectList<K>
+    where
+        K: k8s_openapi::Resource + Clone + DeserializeOwned + Meta,
+    {
+        self.runtime.block_on(async {
+            self.kube_client
+                .list_labeled(label_selector)
+                .await
+                .expect("List of Stackable nodes could not be retrieved")
+        })
+    }
+
+    /// Applies the given custom resource definition and blocks until it is accepted.
+    pub fn apply_crd(&self, crd: &CustomResourceDefinition) {
+        self.runtime.block_on(async {
+            self.kube_client
+                .apply_crd(crd)
+                .await
+                .expect("Custom resource definition coult not be applied")
+        })
+    }
+
+    /// Searches for a named resource.
+    pub fn find<K>(&self, name: &str) -> Option<K>
+    where
+        K: k8s_openapi::Resource + Clone + DeserializeOwned + Meta,
+    {
+        self.runtime
+            .block_on(async { self.kube_client.find::<K>(name).await })
+    }
+
+    /// Applies a resource with the given YAML specification.
+    pub fn apply<K>(&self, spec: &str) -> K
+    where
+        K: k8s_openapi::Resource + Clone + DeserializeOwned + Meta + Serialize,
+    {
+        self.runtime.block_on(async {
+            self.kube_client
+                .apply::<K>(spec)
+                .await
+                .expect("Resource could not be applied")
+        })
+    }
+
+    /// Creates a resource with the given YAML specification.
+    pub fn create<K>(&self, spec: &str) -> K
+    where
+        K: k8s_openapi::Resource + Clone + DeserializeOwned + Meta + Serialize,
+    {
+        self.runtime.block_on(async {
+            self.kube_client
+                .create(spec)
+                .await
+                .expect("Resource could not be created")
+        })
+    }
+
+    /// Deletes the given resource.
+    pub fn delete<K>(&self, resource: K)
+    where
+        K: k8s_openapi::Resource + Clone + DeserializeOwned + Meta,
+    {
+        self.runtime.block_on(async {
+            self.kube_client
+                .delete(resource)
+                .await
+                .expect("Resource could not be deleted")
+        })
+    }
+
+    /// Verifies that the given pod condition becomes true within 30 seconds.
+    pub fn verify_pod_condition(&self, pod: &Pod, condition_type: &str) {
+        self.runtime.block_on(async {
+            self.kube_client
+                .verify_pod_condition(pod, condition_type)
+                .await
+                .expect("Pod condition could not be verified")
+        })
+    }
+}
+
+/// A client for interacting with the Kubernetes API
+///
+/// [`KubeClient`] wraps a [`Client`][kube::Client]. It provides methods
```

Comment:
I believe the Markdown link syntax is wrong here, maybe?

Reply:
It should be correct. ``[`KubeClient`]`` is a shortcut reference link and ``[`Client`][kube::Client]`` is a full reference link, see https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html#detailed-design. For instance, kube-rs uses also these kinds of links: https://github.com/clux/kube-rs/blob/0.52.0/kube/src/client/mod.rs#L3-L7

Unfortunately I was not able to generate the documentation for the tests. Therefore I could not verify that these references are working.